sorbet/rbi/shims/set.rbi

```diff
@@ -0,0 +1,5 @@
+# typed: strict
+
+class Set
+  def compare_by_identity; end
```

Comment:
Good catch, I opened a PR upstream to define this method: https://github.com/sorbet/sorbet/pull/4906.